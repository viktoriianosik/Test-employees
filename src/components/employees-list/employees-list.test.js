import React from "react";
import renderer from "react-test-renderer";
import EmployeesList from "./employees-list";
import * as reactRedux from "react-redux";
import testUsers from "../../test-data/test-data";

describe(`Render component EmployeesList correctly`, () => {
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  const mockedDispatch = jest.fn();
  
  beforeEach(() => {
    useDispatchMock.mockClear()
  });

  it(`Should EmployeesList connected to store render correctly`, () => {
    useDispatchMock.mockReturnValue(mockedDispatch);
    const tree = renderer.create(
      <EmployeesList users={testUsers} selectedUsers={testUsers}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});