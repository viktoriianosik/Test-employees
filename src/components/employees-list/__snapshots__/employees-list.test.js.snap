// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component EmployeesList correctly Should EmployeesList connected to store render correctly 1`] = `
<div
  className="employee"
>
  <h2
    className="employee__title title"
  >
    Employees
  </h2>
  <ul
    className="users-list"
  >
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        A
      </h3>
      <label
        className="user-item__name"
      >
        Asi Bob
        <input
          checked={true}
          className="user-item__input visually-hidden"
          onChange={[Function]}
          type="checkbox"
          value="222"
        />
        <span
          className="user-item__checkbox"
        />
      </label>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        B
      </h3>
      <label
        className="user-item__name"
      >
        Bort Alex
        <input
          checked={true}
          className="user-item__input visually-hidden"
          onChange={[Function]}
          type="checkbox"
          value="111"
        />
        <span
          className="user-item__checkbox"
        />
      </label>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        C
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        D
      </h3>
      <label
        className="user-item__name"
      >
        Dark Anna
        <input
          checked={true}
          className="user-item__input visually-hidden"
          onChange={[Function]}
          type="checkbox"
          value="444"
        />
        <span
          className="user-item__checkbox"
        />
      </label>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        E
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        F
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        G
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        H
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        I
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        J
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        K
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        L
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        M
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        N
      </h3>
      <label
        className="user-item__name"
      >
        Nofl Mark
        <input
          checked={true}
          className="user-item__input visually-hidden"
          onChange={[Function]}
          type="checkbox"
          value="333"
        />
        <span
          className="user-item__checkbox"
        />
      </label>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        O
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        P
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        Q
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        R
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        S
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        T
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        U
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        V
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        W
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        X
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        Y
      </h3>
      <p>
        -----
      </p>
    </li>
    <li
      className="users-item"
    >
      <h3
        className="users__letter"
      >
        Z
      </h3>
      <p>
        -----
      </p>
    </li>
  </ul>
</div>
`;
